codec-http2/src/test/java/io/netty/handler/codec/http2/Http2ConnectionHandlerTest.java

```diff
@@ -199,4 +210,55 @@ public void encoderAndDecoderAreClosedOnChannelInactive() throws Exception {
         verify(encoder).close();
         verify(decoder).close();
     }
+
+    @Test
+    public void writeRstOnNonExistantStreamShouldSucceed() throws Exception {
+        handler = newHandler();
+        handler.writeRstStream(ctx, NON_EXISTANT_STREAM_ID, STREAM_CLOSED.code(), promise);
+        verify(frameWriter, never())
+            .writeRstStream(any(ChannelHandlerContext.class), anyInt(), anyLong(), any(ChannelPromise.class));
+        assertTrue(promise.isDone());
+        assertTrue(promise.isSuccess());
+        assertNull(promise.cause());
+    }
+
+    @Test
+    public void writeRstOnClosedStreamShouldSucceed() throws Exception {
+        handler = newHandler();
+        when(frameWriter.writeRstStream(eq(ctx), eq(STREAM_ID),
+                anyLong(), any(ChannelPromise.class))).thenReturn(future);
+        when(stream.state()).thenReturn(CLOSED);
+        // The stream is "closed" but is still known about by the connection (connection().stream(..)
+        // will return the stream). We should still write a RST_STREAM frame in this scenario.
+        handler.writeRstStream(ctx, STREAM_ID, STREAM_CLOSED.code(), promise);
+        verify(frameWriter).writeRstStream(eq(ctx), eq(STREAM_ID), anyLong(), any(ChannelPromise.class));
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void closeListenerShouldBeNotifiedOnlyOneTime() throws Exception {
+        handler = newHandler();
+        when(connection.activeStreams()).thenReturn(Arrays.asList(stream));
+        when(connection.numActiveStreams()).thenReturn(1);
+        when(future.isDone()).thenReturn(true);
+        when(future.isSuccess()).thenReturn(true);
+        doAnswer(new Answer<ChannelFuture>() {
+            @Override
+            public ChannelFuture answer(InvocationOnMock invocation) throws Throwable {
+                Object[] args = invocation.getArguments();
+                ChannelFutureListener listener = (ChannelFutureListener) args[0];
+                // Simulate that all streams have become inactive by the time the future completes
+                when(connection.activeStreams()).thenReturn(Collections.<Http2Stream>emptyList());
+                when(connection.numActiveStreams()).thenReturn(0);
+                // Simulate the future being completed
+                listener.operationComplete(future);
+                return future;
+            }
+        }).when(future).addListener(any(GenericFutureListener.class));
+        handler.close(ctx, promise);
+        handler.closeStream(stream, future);
+        // Simulate another stream close call being made after the context should already be closed
```

Comment:
I can fix this but I wonder do we need proper punctuation on comments internal to a method block?